scanner/Makefile

```diff
@@ -19,15 +19,40 @@ else
 QUAY_TAG_EXPIRATION=never
 endif
 
+# CGO is disabled for Scanner V4 builds because:
+#
+# 1. Local builds aren't dockerized. And, upon packaging the locally-built scanner in a container when e.g. host
+# OS's glibc does not match version in target container, scanner refuses to start with errors looking like this:
+#    $ /usr/local/bin/scanner --help
+#    /usr/local/bin/scanner: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by /usr/local/bin/scanner)
+#    /usr/local/bin/scanner: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /usr/local/bin/scanner)
+#
+# 2. GitHub builds for arm64, ppc64le and s390x need support for cross-compilation which, when unavailable, makes
+# builds fail with errors like
+#    gcc_arm64.S:30: Error: no such instruction: `stp x29,x30,[sp,'
+#    gcc_arm64.S:34: Error: too many memory references for `mov'
+#
+# This flag must be overridden in commercial/downstream builds in arguments to `make`.
+#    make CGO_ENABLED=1 blah          # Correct - flips it to 1.
+#    CGO_ENABLED=1 make blah          # Wrong! This has no effect. Don't do this.
+#    export CGO_ENABLED=1; make blah  # Wrong! This has no effect. Don't do this.
+# Similarly, don't do the following in Dockerfile
+#    ENV CGO_ENABLED=1                # Wrong!..
+#    RUN make blah                    # because it has no effect. Don't do this.
+# If you need to separate the declaration and the usage in Dockerfile, do this instead
+#    ARG CGO_ENABLED=1
+#    RUN make CGO_ENABLED=$CGO_ENABLED blah
+#
+# TODO: remove this assignment to set CGO_ENABLED to 1 (see env.mk) to minimize testing difference with FIPS builds.
```

Comment:
is there a ticket associated with this?

Reply:
Similarly, do our upstream images need to be FIPS compliant or just downstream?

Reply:
> is there a ticket associated with this? if not, can you create one and link it here?

I created https://issues.redhat.com/browse/ROX-23969 and assigned to the Scanner team. Feel free to take ownership of it and e.g. resolve if there are duplicates.

> Similarly, do our upstream images need to be FIPS compliant or just downstream?

Upstream images don't have to be ready for FIPS and in many cases they likely could not be, but the closer we make them - less difference we need to worry about in testing.